py_zipkin/zipkin.py

```diff
@@ -169,7 +170,17 @@ def __init__(
     def __call__(self, f):
         @functools.wraps(f)
         def decorated(*args, **kwargs):
-            with self:
+            with zipkin_span(
+                service_name=self.service_name,
+                span_name=self.span_name,
+                zipkin_attrs=self.zipkin_attrs,
```

Comment:
Yeah, because it'll essentially be broken (or probably result in odd behavior for anyone using those features, at least) for those args. You're right that we should do this in a followup ticket.